### PR TITLE
Updating datamodel docs with missing parameter fields

### PR DESCRIPTION
**Pull request by aolofsson**

### CODE REVIEW

File: docs/user_guide/data_model.rst [hunk 1]
```diff
@@ -28,28 +28,44 @@ The table below summarizes mandatory fields used in all parameter definitions.
...
+   * - **set**
+     - State indicating if parameter value has been modified by user.
+     - True / False
 
    * - **lock**
      - Enable/disable for set()/add() methods
      - True / False
 
+   * - **scope**
+     - Scope of paremter in schema
```
Comment (nmoroze) [inline docs/user_guide/data_model.rst:66]: typo: parameter

File: docs/user_guide/data_model.rst [hunk 2]
```diff
@@ -58,6 +74,22 @@ The table below summarizes mandatory fields used in all parameter definitions.
...
+   * - **shorthelp**
+     - Short single line help string.
+     - String
+
+   * - **help**
+     - Multi-line documentation string
+     - String
+
+   * - **example**
+     - Usage examples for CLI ad API
```
Comment (nmoroze) [inline docs/user_guide/data_model.rst:86]: typo: and

File: docs/user_guide/data_model.rst [hunk 3]
```diff
@@ -161,59 +192,55 @@ The SiliconCompiler Schema is roughly divided into the following major sub-group
...
+     - 41
      - Design interface specifications
 
    * - **units**
      - 9
      - Global units
 
    * - **total**
-     - 317
+     - 350
```
Comment (nmoroze) [inline docs/user_guide/data_model.rst:243]: hopefully these don't change too much soon :)

File: docs/user_guide/data_model.rst [hunk 4]
```diff
@@ -28,28 +28,44 @@ The table below summarizes mandatory fields used in all parameter definitions.
...
+   * - **nodevalue**
+     - Dictionary of values based on step & index keys
+     - Dictionary
 
-   * - **example**
-     - Usage examples for CLI ad API
-     - String
+   * - **pernode**
+     - Enables/disables setting of value on a per node basis
+     - never, required, optional
```
Comment (nmoroze) [inline docs/user_guide/data_model.rst:55]: maybe quotes around these to indicate they are strings (consistent with "scope" listing)

**Review by nmoroze — APPROVED** [review]
Looks good, left comments on a few typos I caught!

**Comment by aolofsson** [comment]
Thanks for the review!

**Pull request merged**